framework/auth/campaigns.py

```diff
@@ -75,24 +75,49 @@ def get_campaigns():
                     }
                 })
 
+            # Collections
+            collection_providers = CollectionProvider.objects.all()
+            for collection_provider in collection_providers:
+                campaign = f'{collection_provider._id}-collections'
+                system_tag = provider_source_tag(collection_provider._id, 'collection')
+                name = collection_provider.name
+                url_path = f'collections/{collection_provider._id}'
+                newest_campaigns.update({
+                    campaign: {
+                        'system_tag': system_tag,
+                        'redirect_url': furl.furl(DOMAIN).add(path=url_path),
+                        'confirmation_email_template': mails.CONFIRM_EMAIL_COLLECTIONS(name),
+                        'login_type': 'proxy',
+                        'provider': name,
+                    }
+                })
+
             # Proxy campaigns: Registries, OSF only
-            # TODO: refactor for futher branded registries when there is a model for registries providers
-            newest_campaigns.update({
-                'osf-registries': {
-                    'system_tag': provider_source_tag('osf', 'registry'),
-                    'redirect_url': furl.furl(DOMAIN).add(path='registries/').url,
-                    'confirmation_email_template': mails.CONFIRM_EMAIL_REGISTRIES_OSF,
-                    'login_type': 'proxy',
-                    'provider': 'osf',
-                    'logo': settings.OSF_REGISTRIES_LOGO
-                }
-            })
+            registration_providers = RegistrationProvider.objects.all()
+            for registration_provider in registration_providers:
+                if registration_provider._id == 'osf':
+                    url_path = 'registries/'
+                    template = 'osf'
+                else:
+                    url_path = f'registries/{registration_provider._id}'
+                    template = 'branded'
+                campaign = f'{registration_provider._id}-registries'
+                newest_campaigns.update({
+                    campaign: {
+                        'system_tag': provider_source_tag(registration_provider._id, 'registry'),
+                        'redirect_url': furl.furl(DOMAIN).add(path=url_path).url,
+                        'confirmation_email_template': mails.CONFIRM_EMAIL_REGISTRIES(template, registration_provider._id),
+                        'login_type': 'proxy',
+                        'provider': registration_provider.name,
+                        'logo': registration_provider._id if name != 'OSF Registries' else settings.OSF_REGISTRIES_LOGO
```

Comment:
We don't - similar to PreprintProvider's, they'll need to be stored on `osf-assets` (https://github.com/CenterForOpenScience/osf-assets/tree/master/files). It looks like this will also require some modifications to this line on the base notification email template (https://github.com/CenterForOpenScience/osf.io/blob/develop/website/templates/emails/notify_base.mako#L25).

Reply:
I'm planning on modifying https://github.com/CenterForOpenScience/osf.io/blob/develop/website/templates/emails/notify_base.mako#L25 to `https://raw.githubusercontent.com/CenterForOpenScience/osf-assets/master/files/${context.get('logo')}/wide_white.png` and modifying the `logo` to include the directory (so a preprint provider logo would be `/preprints-assets/psyarxiv` and a registration provider logo would be `/registries-assets/egap`). It looks like I would have to change it for unregistered contributor emails in addition to the new user confirmation emails. 

Does that sound reasonable? Would it be better to split this off into another ticket? @brianjgeiger

Reply:
That sounds reasonable. It doesn't seem large enough to warrant another ticket, I don't think.